lib/salus/scanners/slither.rb

```diff
@@ -0,0 +1,94 @@
+require 'json'
+require 'salus/scanners/base'
+
+# Slither (https://github.com/crytic/slither) is a Python-based static
+# analysis tool for Solidity smart contracts
+
+module Salus::Scanners
+  class Slither < Base
+    include Salus::Formatting
+
+    def should_run?
+      # only support truffle/hardhat for now
+      return false if !@repository.package_json_present? || !has_package_config
+
+      @repository.sol_file_present? != false
```

Comment:
Can we just write `@repository.sol_file_present?` as the last line of the method instead of using `@repository.sol_file_present? != false`. In other words, is the following code-snippet same?

```ruby
def should_run?
   ....
   @repository.sol_file_present?
end
```

Reply:
Not really, because `@repository.sol_file_present?` does a wildcard search, which could return an array of file paths.

Reply:
Interesting. In this case, the method name should not end with '?'. It is convention in ruby to use '?' at the end of boolean method which returns true/false values. I will try to get a reference in ruby book later. But, this is an [article](https://medium.com/@sologoubalex/boolean-methods-in-ruby-94a2e907e5ea) on boolean method.